### PR TITLE
Fix Swift 4.2 warnings

### PR DESCRIPTION
**Pull request by wongzigii**

### CODE REVIEW

File: Sources/Nimble/Matchers/Predicate.swift [hunk 1]
```diff
@@ -312,7 +312,7 @@ final public class NMBPredicateStatus: NSObject {
     public static let doesNotMatch: NMBPredicateStatus = NMBPredicateStatus(status: 1)
     public static let fail: NMBPredicateStatus = NMBPredicateStatus(status: 2)
 
-    public override var hashValue: Int { return self.status.hashValue }
+    public override var hash: Int { return self.status.hashValue }
```
Comment (ikesyo) [inline Sources/Nimble/Matchers/Predicate.swift]: Is this compatible with Swift 4.1?

Comment (ikesyo) [reply Sources/Nimble/Matchers/Predicate.swift]: I'll check the warning in Swift 4.2 later.

File: Sources/Nimble/Matchers/HaveCount.swift [hunk 2]
```diff
@@ -7,7 +7,7 @@ import Foundation
 
 /// A Nimble matcher that succeeds when the actual Collection's count equals
 /// the expected value
-public func haveCount<T: Collection>(_ expectedValue: T.IndexDistance) -> Predicate<T> {
+public func haveCount<T: Collection>(_ expectedValue: Int) -> Predicate<T> {
```
Comment (ikesyo) [inline Sources/Nimble/Matchers/HaveCount.swift]: IIRC this had been warned from Swift 4.1, so this is okay (compatible with Swift 4.1).

Comment (ikesyo) [reply Sources/Nimble/Matchers/HaveCount.swift]: https://github.com/apple/swift-evolution/blob/master/proposals/0191-eliminate-indexdistance.md

File: Sources/Nimble/Matchers/Predicate.swift (hunk 1 above)
Comment (wongzigii) [reply Sources/Nimble/Matchers/Predicate.swift]: https://github.com/apple/swift-corelibs-foundation/blob/master/Foundation/NSObject.swift#L326

Comment (wongzigii) [reply Sources/Nimble/Matchers/Predicate.swift]: > `NSObject` implements this by returning `self.hash`. Subclasses can
    customize hashing by overriding the `hash` property.

Comment (wongzigii) [reply Sources/Nimble/Matchers/Predicate.swift]: https://github.com/apple/swift/blob/6fc4cef671947d1d2482951853f7f82854473d91/test/Inputs/clang-importer-sdk/swift-modules-without-ns/ObjectiveC.swift#L77

Comment (ikesyo) [reply Sources/Nimble/Matchers/Predicate.swift]: Totally makes sense :+1: Thanks for the input!

**Pull request merged**